### ISSUE TITLE
Run functional tests: System.AccessViolationException

### ISSUE THREAD

**Issue by ffnoname** [Microsoft/vsts-tasks#2872]
Hey there,

we are permanently running into test runs that get stuck. it happens quite randomly on different test machines, with different configurations. We had those exact same (virtual) machines running our tests with TFS 2012 in a lab environment just fine.

looking into DTAExecutionHost.exe.txt on the test machines, we always find this error message:

> V, 1408, 101, 2016/10/13, 16:15:34.094, 62721708734, DTAExecutionHost.exe, TestRunRequest:SendTestRunMessage: Starting.
I, 1408, 101, 2016/10/13, 16:15:34.094, TestMachineName\DTAExecutionHost, DefaultTestRunEvenHandler.OnTestRunMessage : VstestAdapter.Execution - RunMessage : System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

> I, 1408, 101, 2016/10/13, 16:15:34.094, 62721710791, DTAExecutionHost.exe, TestRunRequest:SendTestRunMessage: Completed.

Followed by endless repeating entries like this.

> V, 1408, 103, 2016/10/13, 16:16:00.688, TestMachineName\DTAExecutionHost, TestExecutionServiceCommandQueueListener.DispatchAsync : Getting Command for Test Agent Microsoft.TeamFoundation.Test.WebApi.TestAgent
Web method running: [http://ourtfsadress:8080/tfs/DefaultCollection/_apis/test/Agents/1766/Commands/1] (GET)1[test]
Web method response: [http://ourtfsadress:8080/tfs/DefaultCollection/_apis/test/Agents/1766/Commands/1] (GET)1[test] 11 ms
V, 1408, 106, 2016/10/13, 16:16:00.688, TestMachineName\DTAExecutionHost, TestExecutionServiceCommandQueueListener:DispatchAsync

Any more info that I should provide?

Thanks!

**Comment by SatishThareja** [comment]
@ffnoname Please share the test execution logs present in 'C:\Dtalogs' and send it to devops_tools@microsoft.com.
Additionally, can you try to execute the tests on fresh VMs and let us know if you hit this issue again?

**Comment by ffnoname** [comment]
hey @SatishThareja. We were able to figure out 2 things:
1.) System.AccessViolationException was caused by our code under test
2.) was also happening with the old environment.

That being sad I think the real issue now is, that the System.AccessViolationException is not letting the test fail properly (in contrast to the old environment) but it somehow puts the test agent into an idle state that it does not come back from. Unfortunately I did not save any logs with that error message and now they have been overridden by new test results. We also have fixed the underlying problem in our code that was causing the Exception. If it comes back I will provide the logs.
